instrumentation/lettuce/lettuce-5.2/src/test/groovy/io/opentelemetry/auto/instrumentation/lettuce/v5_2/LettuceAsyncClientTest.groovy

```diff
@@ -0,0 +1,400 @@
+    age      : "53"
+  ]
+
+  RedisClient redisClient
+  StatefulConnection connection
+  RedisAsyncCommands<String, ?> asyncCommands
+  RedisCommands<String, ?> syncCommands
+
+  def setupSpec() {
+    port = PortUtils.randomOpenPort()
+    incorrectPort = PortUtils.randomOpenPort()
+    dbAddr = HOST + ":" + port + "/" + DB_INDEX
+    dbAddrNonExistent = HOST + ":" + incorrectPort + "/" + DB_INDEX
+    dbUriNonExistent = "redis://" + dbAddrNonExistent
+    embeddedDbUri = "redis://" + dbAddr
+
+    redisServer = RedisServer.builder()
+    // bind to localhost to avoid firewall popup
+      .setting("bind " + HOST)
+    // set max memory to avoid problems in CI
+      .setting("maxmemory 128M")
+      .port(port).build()
+  }
+
+  def setup() {
+    redisClient = RedisClient.create(embeddedDbUri)
+
+    println "Using redis: $redisServer.args"
+    redisServer.start()
+    redisClient.setOptions(CLIENT_OPTIONS)
+
+    connection = redisClient.connect()
+    asyncCommands = connection.async()
+    syncCommands = connection.sync()
+
+    syncCommands.set("TESTKEY", "TESTVAL")
+
+    // 1 set
+    TEST_WRITER.waitForTraces(1)
+    TEST_WRITER.clear()
+  }
+
+  def cleanup() {
+    connection.close()
+    redisServer.stop()
+  }
+
+  def "connect using get on ConnectionFuture"() {
+    setup:
+    RedisClient testConnectionClient = RedisClient.create(embeddedDbUri)
+    testConnectionClient.setOptions(CLIENT_OPTIONS)
+
+    when:
+    ConnectionFuture connectionFuture = testConnectionClient.connectAsync(StringCodec.UTF8,
+      new RedisURI(HOST, port, 3, TimeUnit.SECONDS))
+    StatefulConnection connection = connectionFuture.get()
+
+    then:
+    connection != null
+    // Lettuce tracing does not trace connect
+    assertTraces(0) {}
+
+    cleanup:
+    connection.close()
+  }
+
+  def "connect exception inside the connection future"() {
+    setup:
+    RedisClient testConnectionClient = RedisClient.create(dbUriNonExistent)
+    testConnectionClient.setOptions(CLIENT_OPTIONS)
+
+    when:
+    ConnectionFuture connectionFuture = testConnectionClient.connectAsync(StringCodec.UTF8,
+      new RedisURI(HOST, incorrectPort, 3, TimeUnit.SECONDS))
+    StatefulConnection connection = connectionFuture.get()
+
+    then:
+    connection == null
+    thrown ExecutionException
+    // Lettuce tracing does not trace connect
+    assertTraces(0) {}
+  }
+
+  def "set command using Future get with timeout"() {
+    setup:
+    RedisFuture<String> redisFuture = asyncCommands.set("TESTSETKEY", "TESTSETVAL")
+    String res = redisFuture.get(3, TimeUnit.SECONDS)
+
+    expect:
+    res == "OK"
+    assertTraces(1) {
+      trace(0, 1) {
+        span(0) {
+          operationName "SET"
+          spanKind CLIENT
+          errored false
+          tags {
+            "net.transport" "IP.TCP"
+            "net.peer.ip" "127.0.0.1"
+            "net.peer.port" port
+            "db.type" "redis"
+            "db.statement" "key<TESTSETKEY> value<TESTSETVAL>"
```

Comment:
Can you fill more attributes? https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md